core/modules/layout/layout.admin.inc

```diff
@@ -2431,16 +2436,40 @@ function layout_clone_form_submit($form, &$form_state) {
 }
 
 /**
- * Form callback; Delete a layout.
+ * Form callback. Delete or revert a layout.
  *
  * @ingroup forms
  */
 function layout_delete_form($form, &$form_state, $layout) {
   $form_state['layout'] = $layout;
 
-  if ($layout->storage === LAYOUT_STORAGE_NORMAL) {
+  $delete_layout = ($layout->storage === LAYOUT_STORAGE_NORMAL);
+  if ($delete_layout) {
     $question = t('Delete layout @title?', array('@title' => $layout->title));
-    $text = t('Are you sure you want to delete this layout? Deleting a layout cannot be undone.');
+
+    if (layout_removal_removes_page($layout)) {
+      $path = $layout->getPath();
+      $site_config = config('system.core');
+      if ($path == $site_config->get('site_frontpage')) {
+        $url = url('admin/config/system/site-information') . '#edit-front-page';
+        $text = t('This layout creates a page at path %path that is used as the home page of the site. If you delete or disable this layout, which will entirely remove that page, you must change the home page on the <a href="!url">site information configuration page</a> to a different path.', array('%path' => $path, '!url' => $url));
```

Comment:
Below we say `the page at %path` rather than `the page at path %path`.  I think I prefer the former. 

I also think we can split this up into shorter sentences...  giving that a go:

`This layout creates a page at %path that is currently the home page of this site. If you delete or disable this layout, that page will be removed. You must also change the home page setting on the <a href="!url">site information configuration page</a>. `

Reply:
Changed.